src/gallery/actions/delete-dynamic-background.action.ts

```diff
@@ -9,13 +10,35 @@ export const deleteDynamicBackground = async (
   dynamicBackgroundId: string,
 ): Promise<IDynamicBackground> => {
   try {
-    const response = await fetcher.delete<IDynamicBackground>(
-      StringUtils.bindContext(endpoints.MANAGE_DYNAMIC_BACKGROUND, {
-        id: dynamicBackgroundId,
-      }),
+    const response = await fetcher
+      .delete<IDynamicBackground>(
+        StringUtils.bindContext(endpoints.MANAGE_DYNAMIC_BACKGROUND, {
+          id: dynamicBackgroundId,
+        }),
+      )
+      .catch(() => {
+        throw errorFactory.create(GalleryErr.DELETE_DYNAMIC_BACKGROUND_FAILED);
+      });
+
+    switch (response.status) {
+      case 200:
+        return response.data;
+      case 403:
+        throw errorFactory.create(
+          GalleryErr.DELETE_DYNAMIC_BACKGROUND_FORBIDDEN,
+        );
+      case 404:
+        throw errorFactory.create(
+          GalleryErr.DELETE_DYNAMIC_BACKGROUND_NOT_FOUND,
```

Comment:
```suggestion
          GalleryErr.DYNAMIC_BACKGROUND_NOT_FOUND,
```

Pour les codes d'erreurs de type "not found", il pourrait être intéressant de plutôt faire un code d'erreur par ressource. Ici par exemple la ressource concernée est le "DynamicBackground". On aurait donc un seul code d'erreur pour tout ce qui est "DynamicBackground not found" (donc la route DELETE, PUT et GET incluant un ID de DynamicBackground).